feat: send product flow to approve payment mutation

jscheinhorn opened the pull request:
### Description
This adds a new optional param in the Approve Applepay graphql mutation in order to specify that the applepay flow is coming from the applepay custom component rather than Smart Payment Buttons.

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
ApplePay flow can come from more than one integration. On the SDK we have SPB and Custom Components.
https://paypal.atlassian.net/browse/DTALTPAY-1623

### Reproduction Steps (if applicable)
Merchant specifies `components=applepay` in SDK script to use the paypal-applepay-components custom component.
Selecting the applepay payment method will start the flow. Buyer approval leads to the approve applepay payment mutation call.

### Screenshots (if applicable)
![image](https://github.com/paypal/paypal-applepay-components/assets/29203245/1dbfd6fa-397f-484f-ac02-5ef52cba7f9c)

### Dependent Changes (if applicable)
Changes to xobuyer mutation are for an optional variable, so this is a non-breaking change.

RatnadeepSimhadri reviewed (approved):
Looks Good @jscheinhorn . One Question why did you have to change the ClientID in the Test ?

jscheinhorn commented:
@RatnadeepSimhadri the sandbox test app domain was de-registered from that client ID and changed to the client ID 
 updated in the repo. I learned this from wallet team, but am not sure why the change happened. Client ID will have to be changed for the sb test app going forward.
